app/scripts/controllers/user-storage/services.ts

```diff
@@ -0,0 +1,78 @@
+import encryption from './encryption';
+import { UserStorageEntryKeys, createEntryPath } from './schema';
+
+export const USER_STORAGE_API = process.env.USER_STORAGE_API || '';
+export const USER_STORAGE_ENDPOINT = `${USER_STORAGE_API}/api/v1/userstorage`;
+
+export type GetUserStorageResponse = {
+  HashedKey: string;
+  Data: string;
+};
+
+export type UserStorageOptions = {
+  bearerToken: string;
+  entryKey: UserStorageEntryKeys;
+  storageKey: string;
+};
+
+export async function getUserStorage(
+  opts: UserStorageOptions,
+): Promise<string | null> {
+  const path = createEntryPath(opts.entryKey, opts.storageKey);
+  const url = new URL(`${USER_STORAGE_ENDPOINT}${path}`);
+
+  const encryptedData = await fetch(url.toString(), {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${opts.bearerToken}`,
+    },
+  })
+    .then((r) => {
+      if (r.status === 404) {
+        // Acceptable Error
+        return null;
+      }
+      if (r.status !== 200) {
+        throw new Error('Unable to get User Storage');
+      }
+      return r.json();
+    })
+    .then((r: GetUserStorageResponse | null) => r?.Data)
+    .catch(() => null);
+
+  if (!encryptedData) {
+    return null;
+  }
+
+  try {
+    const decryptedData = encryption.decryptString(
+      encryptedData,
+      opts.storageKey,
+    );
+    return decryptedData;
+  } catch {
+    return null;
+  }
+}
+
```

Comment:
The discussion on error handling patterns that happened in the other PR aside, I notice that two different patterns are used in each of the functions in this file. One does `await fetch` followed by a `then`, within which the response is handled, and the other does not chain do `then` chaining. Why is that?

Reply:
Aha, the `encryption.decryptString` is synchronous so we are unable to await or `.then` here.

I think since we are using try-catch anyways here I'll update the methods to stick with 1 global try catch.

Reply:
This has been refactored, 1 try-catch :)